### ISSUE TITLE
modifies-parameter: improve detection for slices.Delete and slices.DeleteFunc

### ISSUE THREAD

**Issue by PascalBourdier** [mgechev/revive#1410]
**Is your feature request related to a problem? Please describe.**
if we use `slices.DeleteFunc` in a func, input slice is updated silently

**Describe the solution you'd like**
warn us about `slices.Delete` and `slices.DeleteFunc` usage on input parameters

**Describe alternatives you've considered**
N/A

**Additional context**
https://github.com/golang/go/issues/66141
https://go.dev/play/p/sQJry0QkIs_o

**Comment by alexandear** [comment]
Thanks for the suggestion!

Just to clarify: are you proposing that the `modifies-parameter` rule should report any usage of `slices.Delete` / `slices.DeleteFunc` when the result is assigned back to the input parameter?

For example, should this be reported?

```go
func f(s []int) {
	s = slices.DeleteFunc(s, someCond)
}
```

But this one ignored?

```go
func f(s []int) {
	_ = slices.DeleteFunc(s, someCond)
}
```

Want to make sure we align on the scope of the rule enhancement.

**Comment by PascalBourdier** [comment]
this one should be reported too, because `s` is updated !
```go
func f(s []int) {
	_ = slices.DeleteFunc(s, someCond)
}
```

2 following cases should be ignored

`clone` case: 
```go
func f(s []int) {
	_ = slices.DeleteFunc(slices.Clone(s), someCond)
}
```
`copy` case:
```go
func f(s []int) {
	var c []int
	_ = copy(s, c)
	_ = slices.DeleteFunc(c, someCond)
}
```

**Comment by PascalBourdier** [comment]
IMO, it could be a better practice to add a "skip rule" explicitly for the case:
```go
//revive:disable:modifies-parameter
func f(s []int) {
	_ = slices.DeleteFunc(s, someCond)
}
//revive:enable:modifies-parameter
```
